src/mcnp/mcnp_funcs.cpp

```diff
@@ -251,33 +254,33 @@ void write_cell_cards(std::ostringstream& lcadfile, const char* mcnp_version_maj
       } else if (mcnp_version_major[0] == '6') {
         mcnp_name = pyne::particle::mcnp6(particle_name);
       } else {
-        std::cout << "Unknown MCNP verison: " << mcnp_version_major << std::endl;
+        std::cerr << "Unknown MCNP verison: " << mcnp_version_major << std::endl;
         exit(EXIT_FAILURE);
       }
       double imp = 1.0;
       // if we find graveyard always have importance 0.0
-      if (mat_name.find("Graveyard") != std::string::npos) {
+      if (mat_name.find(graveyard_str) != std::string::npos) {
         imp = 0.0;
         // no splitting can happenin vacuum set to 1
-      } else if (mat_name.find("Vacuum") != std::string::npos) {
+      } else if (mat_name.find(vacuum_str) != std::string::npos) {
         imp = 1.0;
         // otherwise as the map says
       } else {
         imp = DMD->importance_map[entity][particle_name];
       }
-      importances += "imp:" + mcnp_name + "=" + _to_string(imp) + " ";
+      importances += "imp:" + mcnp_name + "=" + std::to_string(imp) + " ";
```

Comment:
It seems that `std::to_string`  is always going to supply 6 decimal places for floating-point types, even if they aren't needed. The test output of `dagmcnp_unit_tests` expects the lcad output to have integer values for the neutron/photon importances and it's causing CI to fail.

We can either change the expected test output or we could get these values as integers before writing them to the lcad file. Given that MCNP treats these importances as floating-point values, I'd say we should update the expected results in our test suite. I'll defer to @makeclean on this though as he's far more knowledgable than I :)

Reply:
[mcnp_funcs.cpp actually defines the importances as doubles.](https://github.com/svalinn/DAGMC/blob/develop/src/mcnp/mcnp_funcs.cpp#L257) We _could_ define it as an int . If the input to `to_string()` is an int, then there's no trailing zeros. But this is probably not recommended because users can define an importance as a doubles.

I am going to update the tests to expect doubles there.

Reply:
Wasn't saying that `mcnp_funcs.cpp` defines the importance as an integer.. just that the expected test output has something that looks like an int. :) In the end, the important part is how MCNP interprets them when determining how we should write them IMO, so I back your call to update the expected test output to expect doubles.

Reply:
Right- Sorry that was my own processing of what was wrong because I was confused by the trailing zeros (I assumed the importances were defined as ints, but doubles makes sense). Anyway, tests have been updated.

Reply:
Nice! Only caveat here is that importances will now be truncated to 6 digits... something we should probably be aware of. The [example here](https://devdocs.io/cpp/string/basic_string/to_string) explains in more detail. 

After reading a bit on this subject, it seems that our method in `mcnp_funds.cpp` for higher precision double --> string conversion (with the addition of a `std::setprecision` in the stream) is still recommended. :man_shrugging:

I'd be surprised if more than 6 digits are used for these values very often, so I'm good w/ whatever you decide.

Reply:
I'll look into the `std::setprecision` option. If we set the precision to match doubles, that's 15 decimal points.

The other thing I just noticed given your example is that `to_string()` won't write out values in scientific notation. So if a value is 1e-3, then it will be written as 0.00100. If a value is smaller than 1e-6, then it will return 0.000000. So that concerns me too. If we set precision to 15, that allows for numbers as small as 1e-15. That doesn't seem like a good idea to limit the range like that.

What do you think? Now I am thinking that given the limitations of this `to_string()` function that we should actually keep what we had.

Reply:
@gonuke or @makeclean - any thoughts on this?

Reply:
I agree that the `to_string()` is not a good change

Reply:
I agree as well -- we should go back to our `_to_string` function for floating point values. We can still use `std::to_string` for `int`'s like material number though. Probably best to use the standard library where we can.

Reply:
This sounds like a good idea- I will update accordingly.

I think `to_string()` will still convert ints to floats to write out (eq, `1` will become `1.000000` by default). But I might be able to play with the precision to avoid that.

Reply:
You'll find that `std::to_string()` will write ints without trailing zeros. Its implementation is specialized depending on the type passed to it. So no extra work should be needed there :tada: